src/pages/index.js

```diff
@@ -1,105 +1,284 @@
+      avatarEditForm.reset();
+    });
+  }
+  handleSubmit(makeRequest, avatarEditFormPopup);
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getCardElement();
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  cardSection.addItem(cardElement);
+function handleNewItemSubmit(inputValues) {
+
+  function makeRequest() {
+    return api.addNewCard(inputValues.title, inputValues.link)
+    .then((newCardData) => {
+      renderCard(newCardData);
+      newItemPopup.close();
+      newItemForm.reset();
+    });
+  }
+  handleSubmit(makeRequest, newItemPopup);
+
 }
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
+function handleLikeCard(cardData) {
+  api
+    .likeCard(cardData._id)
+    .then(() => {
+      cardData.setCardLike(true);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+}
 
-cardSection.renderItems();
+function handleUnlikeCard(cardData) {
+  api
+    .unlikeCard(cardData._id)
+    .then(() => {
+      cardData.setCardLike(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+}
 
-const newItemPopup = new PopupWithForm({
-  popupSelector: "#new-item-modal",
-  handleFormSubmit: handleNewItemSubmit,
+function handleDeleteModal(cardData) {
+  formValidators["delete-form"].resetValidation();
+  deleteCardPopup.open();
+  deleteCardPopup.setFormSubmitHandler(() => {
+    api
+      .deleteCard(cardData._id) //replaced cardId with cardData._id
+      .then(() => {
+        cardData.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch(console.error);
+  });
+
+}
+
+function handleImageClick(cardData) {
+  previewImagePopup.open(cardData);
+}
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Listeners                              */
+/* -------------------------------------------------------------------------- */
+
+profileEditFormPopup.setEventListeners();
+
+profileEditButton.addEventListener("click", function () {
+  const { name, description } = profileUserInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = description;
+  formValidators["profile-edit-form"].resetValidation();
+  formValidators["profile-edit-form"].disableButton();
+  profileEditFormPopup.open();
 });
+
+avatarEditFormPopup.setEventListeners();
+
+avatarEditButton.addEventListener("click", () => {
+  formValidators["avatar-edit-form"].resetValidation();
+  formValidators["avatar-edit-form"].disableButton();
+  avatarEditForm.reset();
+  avatarEditFormPopup.open();
+})
+
 newItemPopup.setEventListeners();
 
 newItemButton.addEventListener("click", () => {
+  formValidators["new-item-form"].resetValidation();
+  formValidators["new-item-form"].disableButton();
+  newItemForm.reset();
```

Comment:
* You reset the form in `then`, so you need to call `disableButton` also in `then`. You need to **disable** the button only in block `then` after a **successful response** to be able to click it again if there is a server error.


```js

newItemButton.addEventListener("click", () => {
  newItemPopup.open();
});

```